app/components-react/hooks/useTree.tsx

```diff
@@ -0,0 +1,71 @@
+import { EventDataNode, DataNode } from 'antd/lib/tree';
+
+export interface IOnDropInfo {
+  node: EventDataNode;
+  dragNode: DataNode;
+  dropPosition: number;
+  dropToGap: boolean;
+}
+
+export function useTree(onlyLeaves?: boolean) {
```

Comment:
it currently is yea, the reason i framed it as a hook was because the current implementation of the tree in this port is just about the most simple version, so i wanted to give room to expand this hook as necessary if for example in the source selector some sort of other functionality is needed